synapse/rest/client/v1/login.py

```diff
@@ -192,10 +193,7 @@ async def _do_other_login(self, login_submission):
                 raise SynapseError(400, "Invalid thirdparty identifier")
 
             if medium == "email":
-                # For emails, transform the address to lowercase.
-                # We store all email addreses as lowercase in the DB.
-                # (See add_threepid in synapse/handlers/auth.py)
-                address = address.lower()
+                address = canonicalise_email(address)
```

Comment:
for the record, I have checked that there are no users on `matrix.org` that will be locked out by this change (there are no users with non-ascii characters in their email addresses). I'm reasonably happy to assume that if it's not a problem on `matrix.org`, it's not a problem anywhere.